src/asm/fstack.cpp

```diff
@@ -243,7 +243,7 @@ bool yywrap(void)
 			// Avoid arithmetic overflow runtime error
 			uint32_t forValue = (uint32_t)contextStack->forValue +
 				(uint32_t)contextStack->forStep;
-			contextStack->forValue = forValue >= 0 ? (int32_t)forValue
+			contextStack->forValue = forValue > 0 ? (int32_t)forValue
```

Comment:
This could use more scrutiny.

Reply:
cc @aaaaaa123456789

Reply:
This line has been nonsense all along. `forValue` is unsigned, and thus it cannot be negative in the first place! I'm not sure of what this is intending to do, but it's not doing it, and the "fix" is nonsense.

Reply:
It comes from #1222, where it was replacing `contextStack->forValue += contextStack->forStep;`.

Reply:
The intention is to assign the unsigned `uint32_t forValue` to the signed `int32_t contextStack->forValue`, in the "obvious" way" (reinterpreting the bit pattern, so high unsigned values are negative two's-complement values).

That's a problem I was recently addressing in another program:
```c
// given: int64_t total, uint64_t value
uint64_t sampleThreshold = 1UL << (bitsPerSample - 1);
...
total += value < sampleThreshold ? value : -((int64_t)~value & (sampleThreshold - 1)) - 1; // convert unsigned to signed
```

So I think the correct check here is:

```c
contextStack->forValue = forValue <= INT32_MAX ? forValue : -(int32_t)~forValue - 1;
```

@aaaaaa123456789 please confirm?